### ISSUE TITLE
CI formatting check

### ISSUE THREAD

**Issue by BNAndras** [exercism/dart#591]
Currently, https://github.com/exercism/dart/blob/main/.github/workflows/test.yml checks the formatting before running the test suite, potentially failing before the test suite is checked. This should instead be done as a separate GH action. 

Any of the three dart files (stub, test suite, or example solution) could be formatted incorrectly so it's unclear which file is the issue from the CI logs. The GH action should specify if possible which file was reformatted and ideally what the change is. `dart format` can show the formatted code for the changed file so perhaps we can compare the formatted code to the original file and display the difference. Surfacing that difference should reduce confusion since we're using dart format with an override for line length. I ran afoul of that early on since locally `dart format` was working fine as far as I could tell.

**Comment by vaibhavsahni009** [comment]
I agree with your point, that would be ideal if we show what line the issue is and show a diff. Maybe it is even better if we can ask the formattor to do the job automatically and raise a commit in the same PR with correct formatting through GHA.

I am willing to work on either scenario, but even my solution would show logs similar to how it is below, and you mentioned file was not mentioned in logs. Are we talking about different logs here?

![image](https://github.com/user-attachments/assets/5d6969b8-bd18-4423-b998-fca01a5988bd)
https://github.com/exercism/dart/actions/runs/10983047051/job/30491989205

**Comment by Stargator** [comment]
The reason the format check is in the test action is to reduce the load of actions that run per PR. But if exercism is okay with it, then I'll separate it out to it's own action.

As far as the output, that can be addressed in another PR.
